pkg/sql/alter_table_set_schema.go

```diff
@@ -80,7 +80,7 @@ func (p *planner) AlterTableSetSchema(
 	for _, dependent := range tableDesc.DependedOnBy {
 		if !dependent.ByID {
 			return nil, p.dependentViewError(
-				ctx, tableDesc.TypeName(), tableDesc.Name,
+				ctx, string(tableDesc.TypeName()), tableDesc.Name,
```

Comment:
`TypeName()` now returns a `catalog.DescriptorType` which aliases `string`, for extra type safety.